catroid/src/androidTest/java/org/catrobat/catroid/test/scratchconverter/protocol/JobHandlerSuccessfulOnJobMessageTest.java

```diff
@@ -0,0 +1,195 @@
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.scratchconverter.protocol;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import com.google.android.gms.common.images.WebImage;
+
+import org.catrobat.catroid.common.Constants;
+import org.catrobat.catroid.scratchconverter.Client;
+import org.catrobat.catroid.scratchconverter.ClientException;
+import org.catrobat.catroid.scratchconverter.protocol.Job;
+import org.catrobat.catroid.scratchconverter.protocol.JobHandler;
+import org.catrobat.catroid.scratchconverter.protocol.message.job.JobAlreadyRunningMessage;
+import org.catrobat.catroid.scratchconverter.protocol.message.job.JobFailedMessage;
+import org.catrobat.catroid.scratchconverter.protocol.message.job.JobFinishedMessage;
+import org.catrobat.catroid.scratchconverter.protocol.message.job.JobOutputMessage;
+import org.catrobat.catroid.scratchconverter.protocol.message.job.JobProgressMessage;
+import org.catrobat.catroid.scratchconverter.protocol.message.job.JobReadyMessage;
+import org.catrobat.catroid.scratchconverter.protocol.message.job.JobRunningMessage;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.AdditionalMatchers;
+import org.mockito.stubbing.Answer;
+
+import java.util.Date;
+
+import static junit.framework.Assert.assertEquals;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+@RunWith(AndroidJUnit4.class)
+public class JobHandlerSuccessfulOnJobMessageTest {
+
+	private static final long JOB_ID_OF_JOB_HANDLER = 1;
+	private Client.ConvertCallback convertCallbackMock;
+	private Job jobSpy;
+	private JobHandler jobHandler;
+
+	@Before
+	public void setUp() throws Exception {
+		jobSpy = spy(new Job(JOB_ID_OF_JOB_HANDLER, "My program", null));
+		convertCallbackMock = mock(Client.ConvertCallback.class);
+		jobHandler = new JobHandler(jobSpy, convertCallbackMock);
+	}
+
+	@After
+	public void tearDown() {
+		verifyNoMoreInteractions(convertCallbackMock);
```

Comment:
this should be part of the test, teardown is for cleanup work, this seems to be integral part of the tests.